src/components/GearForm.tsx

```diff
@@ -0,0 +1,307 @@
+    }
+    return hasMatch;
+  }
+
+  function pushArray(tag: string): void {
+    const tempArray = filters;
+    const filter = new Filter({
+      name: tag,
+      toggle: false
+    });
+    tempArray.push(filter);
+    setFilters(tempArray);
+  }
+
+  function checkViewExists(tag: string): boolean {
+    let hasMatch = false;
+    for (let index = 0; index < viewFilters.length; ++index) {
+      if (viewFilters[index].name === tag) {
+        hasMatch = true;
+      }
+    }
+    return hasMatch;
+  }
+
+  function pushViewArray(tag: string): void {
+    const viewArray = viewFilters;
+    const filter = filters.find(function (filter) {
+      return filter.name === tag;
+    });
+    if (filter) {
+      viewArray.push(filter);
+    }
+  }
+
+  function cleanTag(tag: string): string {
+    tag = tag.trim();
+    const words = tag.split(" ");
+    tag = "";
+    for (let index = 0; index < words.length; ++index) {
+      const word = words[index].charAt(0).toUpperCase() + words[index].slice(1);
+      tag = tag + word + " ";
+    }
+    return tag.trim();
+  }
+
+  function filterItems(gear: Gear[], filters: Filter[]): Gear[] | undefined {
+    const tempArray: Gear[] = [];
+    let queriedGear: Gear[] = [];
+    const activeFilters = filters.filter(function (filter) {
+      return filter.toggle;
+    });
+    if (searchString !== "") {
+      const queries = searchString.split(",");
+      for (let i = 0; i < gear.length; ++i) {
+        if (
+          queries.some(function (query) {
+            return (
+              gear[i].title
+                .toLowerCase()
+                .includes(query.toLowerCase().trim()) ||
+              gear[i].tags.toLowerCase().includes(query.toLowerCase().trim())
+            );
+          })
+        ) {
+          queriedGear.push(gear[i]);
+        }
+      }
+    } else {
+      queriedGear = gear;
+    }
+    for (let i = 0; i < queriedGear.length; ++i) {
+      let hasMatch = false;
+      if (matchAny) {
+        if (
+          activeFilters.some(function (filter) {
+            return queriedGear[i].tags
+              .toLowerCase()
+              .includes(filter.name.toLowerCase());
+          })
+        ) {
+          hasMatch = true;
+        }
+      } else {
+        if (
+          activeFilters.every(function (filter) {
+            return queriedGear[i].tags
+              .toLowerCase()
+              .includes(filter.name.toLowerCase());
+          })
+        ) {
+          hasMatch = true;
+        }
+      }
+      if (hasMatch) {
+        tempArray.push(queriedGear[i]);
+      }
+    }
+
+    return tempArray;
+  }
+
+  const toggleFilter = (filter: Filter): void => {
+    filter.toggle = !filter.toggle;
+    const tempArray = [];
+    for (let index = 0; index < filters.length; ++index) {
+      if (filters[index].name == filter.name) {
+        tempArray.push(filter);
+      } else {
+        tempArray.push(filters[index]);
+      }
+    }
+    setFilters(tempArray);
+  };
```

Comment:
This one is another place where you just need to create a dictionary.  No looping needed.
You test if `filters[name]` exists.  That's it.  Easy peasy.